network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/extensions/RemoteReactivePowerControlImpl.java

```diff
@@ -99,9 +99,11 @@ public RemoteReactivePowerControl setRegulatingTerminal(Terminal regulatingTermi
     public void onReferencedRemoval(Terminal removedTerminal) {
         // we cannot set regulating terminal to null because otherwise extension won't be consistent anymore
         // we cannot also as for voltage regulation fallback to a local terminal
-        // so we just remove the extension
+        // so we just remove the extension only if it is the regulated terminal
         LOGGER.warn("Remove 'RemoteReactivePowerControl' extension of generator '{}', because its regulating terminal has been removed", getExtendable().getId());
-        getExtendable().removeExtension(RemoteReactivePowerControl.class);
+        if (removedTerminal != null && removedTerminal.equals(getRegulatingTerminal())) {
```

Comment:
if fixes a bug that is detected with a new tck test so it can be fixed in another pr, what do you think @etiennehomer

Reply:
Ok for me !